python/clx/tests/test_dga_detector.py

```diff
@@ -37,40 +33,47 @@
 )
 dataset = DetectorDataset(test_df, test_batchsize)
 
-# download model
-if not path.exists(MODEL_FILENAME):
-    fs = s3fs.S3FileSystem(anon=True)
-    fs.get(S3_BASE_PATH + "/" + MODEL_FILENAME, MODEL_FILENAME)
+dd = DGADetector()
+dd.init_model()
+
+
+def test_train_model():
+    if torch.cuda.is_available():
+        total_loss = dd.train_model(dataset)
+        assert isinstance(total_loss, (int, float))
 
 
-def test_load_model():
+def test_evaluate_model():
     if torch.cuda.is_available():
-        dd = DGADetector()
-        dd.load_model(MODEL_FILENAME)
-        assert isinstance(dd.model, RNNClassifier)
+        accuracy = dd.evaluate_model(dataset)
+        assert isinstance(accuracy, (int, float))
 
 
 def test_predict():
     if torch.cuda.is_available():
-        dd = DGADetector()
         test_domains = cudf.Series(["nvidia.com", "dfsdfsdf"])
-        dd.load_model(MODEL_FILENAME)
-        actual_output = dd.predict(test_domains)
-        expected_output = cudf.Series([1, 0])
-        assert actual_output.equals(expected_output)
+        preds = dd.predict(test_domains)
+        assert isinstance(preds, cudf.core.series.Series)
```

Comment:
Would add just a few extra checks if these apply
```
assert len(preds) == 2
assert preds.dtype == int
```

Reply:
updated. thank you!